File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    vector<int> A(9);
    int sum = 0;


    for (int i = 0; i < 9; i++) {
        cin >> A[i];
        sum += A[i];
    }


    bool found = false;
    for (int i = 0; i < 9; i++) {
        for (int j = i + 1; j < 9; j++) {
            if (sum - A[i] - A[j] == 100) {
                A[i] = A[j] = -1; 
                found = true;
                break;
            }
        }
        if (found) break;
    }

    sort(A.begin(), A.end());

 
    for (int i = 0; i < 9; i++) {
        if (A[i] != -1) {
            cout << A[i] << "\n";
        }
    }

    return 0;
}

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.java

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int[] arr = new int[9];

		int sum=0;

		for(int i=0;i<9;i++) {
			int n = Integer.parseInt(br.readLine());
			arr[i] = n;
			sum += arr[i];
		}

		Arrays.sort(arr);

		int a=0,b=0;

		for(int i=0;i<arr.length-1;i++) {

			for(int j=i+1;j<arr.length;j++) {

				if(sum - arr[i] - arr[j] == 100) {
					a=i;
					b=j;
					break;
				}
			}
		}

		for(int i=0;i<arr.length;i++) {

			if(i!=a && i!=b) {
				System.out.println(arr[i]);
			}
		}
	}

}
